compose/composer.go

```diff
@@ -2,50 +2,82 @@ package compose
 
 import (
 	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"time"
 )
 
+const lenSpans = 7
+
 func Compose(numErrors, numTransactions, numSpans, numFrames int) []byte {
+	rand.Seed(time.Now().UnixNano())
 	var buf bytes.Buffer
 
 	buf.Write(Metadata)
 	buf.WriteByte('\n')
 
-	stacktrace := make([]byte, len(StacktraceFrame))
-	copy(stacktrace, StacktraceFrame)
-	frames := multiply([]byte(`"stacktrace"`), stacktrace, numFrames)
-
-	span := make([]byte, len(SingleSpan))
-	copy(span, SingleSpan)
-	span = bytes.Replace(span, []byte(`"stacktrace": [],`), frames, -1)
+	for i := 0; i < numTransactions; i++ {
+		ev := randomized("transaction", 8)
+		write(&buf, ndjsonWrapObj("transaction", ev))
 
-	transaction := ndjsonWrapObj("transaction", SingleTransaction)
-	span = ndjsonWrapObj("span", span)
+		for i := 0; i < numSpans; i++ {
+			ev := randomized(fmt.Sprintf("span%d", rand.Intn(lenSpans)+1), 8)
```

Comment:
when you call `rand.Intn(lenSpans)` `numSpans` times, you will likely get duplicated results, is that what you want?
if you want random numbers without repetition, you can do something like:
```go
shuffled := rand.Perm(numSpans)
for i := 0; i < numSpans; i++ {
    ev := randomized(fmt.Sprintf("span%d", shuffled[i], 8)
}
```

Reply:
I think randomizing them is ok, especially since the `numSpans` could be larger than the available span bodies anyways.